docs/snowpark/snowpark-backend-joins.md

```diff
@@ -0,0 +1,178 @@
+      dataSetBColumns.id.alias("y")
+    )
+  }
+```
+
+In Snowpark, when employing a join operation, it's important to note that columns with null values may be encountered. Consequently, you might observe that the select projection simply yields a basic column that could potentially include null values.
+
+### Multiple Joins
+
+In Morphir, you have the flexibility to use two consecutive joins to combine data from three different sources. To achieve this, you can employ one join after another and craft a projection to obtain the desired data. This approach allows you to seamlessly integrate information from multiple datasets, enhancing the versatility of your data manipulation operations, for example:
+
+```elm
+multipleJoin : List TypeA -> List TypeB -> List TypeC -> List {idA: Int, idB: Int, idC: Int}
+multipleJoin dataSetA dataSetB dataSetC = 
+    dataSetA 
+        |> innerJoin dataSetB (\a b -> a.id ==  b.id ) 
+        |> innerJoin dataSetC  (\(x, y) z -> x.id == z.id )
+    |> List.map (\ ((a, b), c) ->
+        { idA = a.id
+        , idB = b.id
+        , idC = c.id
+        })
+```
+
+The transformation to Snowpark will be:
+
+```Scala
+ def multipleJoin(
+    dataSetA: com.snowflake.snowpark.DataFrame
+  )(
+    dataSetB: com.snowflake.snowpark.DataFrame
+  )(
+    dataSetC: com.snowflake.snowpark.DataFrame
+  )(
+    implicit sfSession: com.snowflake.snowpark.Session
+  ): com.snowflake.snowpark.DataFrame = {
+    val dataSetAColumns: deparments.Joins.TypeA = new deparments.Joins.TypeAWrapper(dataSetA)
+    
+    val dataSetBColumns: deparments.Joins.TypeB = new deparments.Joins.TypeBWrapper(dataSetB)
+    
+    val dataSetCColumns: deparments.Joins.TypeC = new deparments.Joins.TypeCWrapper(dataSetC)
+    
+    dataSetA.join(
+      dataSetB,
+      (dataSetAColumns.id) === (dataSetBColumns.id),
+      "inner"
+    ).join(
+      dataSetC,
+      (dataSetAColumns.id) === (dataSetCColumns.id),
+      "inner"
+    ).select(
+      dataSetAColumns.id.alias("idA"),
+      dataSetBColumns.id.alias("idB"),
+      dataSetCColumns.id.alias("idC")
+    )
+  }
+```
+
+In Snowpark, you can perform the union of two or more join functions, enabling support for multiple joins within a single query. In the final part of the query, a projection is generated using a select statement, which selects the desired columns to be included in the resulting DataFrame.
+
+### Requirements 
+
+It is essential to note that for these patterns to be transformed, they must meet certain requirements:
+
+1. Return a Projection as a List of a Record:
+    
+- The join pattern should result in a projection that is a list of a record.
+
+2. Function Parameters Representing a Table:
+    
+- The parameters of the function must be a list of records representing a table. For more information, refer to the Snowpark backend types section.
+
+3. Supported Join Functions:
+    
+- Currently, the Snowpark backend supports the innerJoin and leftJoin functions from Morphir.SDK.
```

Comment:
Add links to Morphir SDK docs https://package.elm-lang.org/packages/finos/morphir-elm/latest/Morphir-SDK-List#leftJoin